test/functional/tests/io_class/test_io_class_stats_file_size_core.py

```diff
@@ -0,0 +1,143 @@
+
+                with TestRun.step("Run fio again with direct=1 and check that "
+                                  "for direct IO class statistics increases."):
+                    [core.reset_counters() for core in cores]
+                    run_io(cores, size_min, size_max, direct=True)
+                    check_statistics(cores, io_classes, io_class, direct=True)
+                    [fs_utils.remove(
+                        f"{core.mount_point}/*", force=True, recursive=True) for core in cores]
+
+                    size_min = size_max + Unit.Byte
+                    if size_max >= Unit.GibiByte:
+                        size_max += 100 * Size(1, Unit.MebiByte)
+                    else:
+                        size_max *= 4
+
+
+def run_io(cores, size_min, size_max, direct=False):
+    fio = Fio().create_command() \
+        .io_engine(IoEngine.libaio) \
+        .do_verify() \
+        .time_based() \
+        .run_time(timedelta(seconds=64) if size_max <= Unit.GibiByte else timedelta(seconds=200)) \
+        .verification_with_pattern() \
+        .read_write(ReadWrite.randrw) \
+        .write_percentage(100) \
+        .io_depth(16) \
+        .block_size(Size(1, Unit.Blocks4096), Size(1, Unit.Blocks512)) \
+        .direct(direct)
+    for core in cores:
+        fio.add_job() \
+            .file_size(size_min) \
+            .target(
+            f"{core.mount_point}/{round(size_min.get_value())}{'_direct' if direct else ''}")
+        fio.add_job() \
+            .file_size(size_max) \
+            .target(
+            f"{core.mount_point}/{round(size_max.get_value())}{'_direct' if direct else ''}")
+    return fio.fio.run()
+
+
+def check_statistics(cores, io_classes, tested_io_class, direct=False):
+    for core in cores:
+        TestRun.LOGGER.info(f"Checking IO class stats for cache {core.cache_id}.")
+        for io_class in io_classes:
+            stat = core.get_io_class_statistics(io_class.id).request_stats.requests_total
+            # when direct == true it's ok if there either are or aren't operations for tested class
+            if (tested_io_class.id == io_class.id and direct) or 'metadata' in io_class.rule:
+                TestRun.LOGGER.info("Check skipped for tested IO Class with direct IO "
+                                    "and for metadata class.")
+                TestRun.LOGGER.info(
+                    f"Total requests for IO Class {io_class.id} {io_class.rule}: {stat}")
+            elif tested_io_class.id == io_class.id or (direct and 'direct' in io_class.rule):
+                if stat <= 50:
+                    TestRun.LOGGER.error(f"Total requests too low "
+                                         f"for IO Class {io_class.id} {io_class.rule}: {stat}")
+                else:
+                    TestRun.LOGGER.info(f"Total requests "
+                                        f"for IO Class {io_class.id} {io_class.rule}: {stat}")
+            elif stat > 50:
```

Comment:
Why is 50 requests number hard coded? It does not follow from the function context.

Reply:
I agree with @pdebski21. I would change fio to generate some more deterministic workload and verify whether cas stats are more or less correct. Or you can go the other way and instead of modifying fio check how many requests was issued and then verify if cas stats are correct